### DISCUSSION TITLE
how to configure qjackctl for Jamulus output

### DISCUSSION THREAD

**Discussion by rdtennent** [jamulussoftware/jamulus#3385] [Technical Support & Help]
I'm running Jamulus on Linux using a "snoball" USB microphone, which I can configure qjackctl to access.  But I'm not getting any Jamulus sound through the earphones.  Other apps work OK through the earphones. How do I configure qjackctl to direct Jamulus-generated sound to the headphones?

**Comment by rdica** [comment]
While Jamulus is running, open a terminal and type these commands `jack_lsp system` and `jack_lsp -c`.
Please paste the output of both commands. Your output should similar to the following:
```
user@host: jack_lsp system
system:capture_1
system:capture_2
system:capture_3
system:capture_4
system:playback_1
system:playback_2
system:playback_3
system:playback_4
````

Basically you need to determine what jack ports your headphone use, then connect jamulus ports to those.
In my configuration, headphones are `system:playback_3` and `system:playback_4` , and Jamulus ports `Jamulus:output-left` and `Jamulus:output-right` are connected to those ports.

**Comment by rdtennent** [comment]
Thanks.  Unfortunately, too late for the scheduled jam session today.  I
expect to get a rpi4B and a hifiberry sound card which I've used in the
past before the next scheduled jam. Shouldn't these instructions be
somewhere on the Jamulus site?

**Comment by rdtennent** [comment]
host~> jack_lsp system
system:capture_1
host:~> jack_lsp -c
system:capture_1
   Jamulus:input left
Jamulus:input left
   system:capture_1
Jamulus:input right
Jamulus:output left
Jamulus:output rig

**Comment by rdica** [comment]
You don't appear to have any system:playback ports. Those are required in order for jack to route audio to speakers or headphones.
Can you detail your OS, and audio hardware?

**Comment by rdtennent** [comment]
It's a Lenovo X220 ThinkPad.  I'm using a USB microphone (which qjackctl
recognizes) and wired headphones through the earphone port on the side.
The headphone works for conventional sources such as YouTube. Running
Linux MInt 21.3.

**Reply by rdica** [reply]
Audio hardware = soundcard/audio interface.
Are you using an audio interface of some type, plugged in with USB?
Are you using the laptops onboard soundcard?

I think your soundcard/audio interface isn't being used by `jackd`, and that you may have jack misconfigured. Your microphone being a USB device would be autodetected as a sound card that has one input and is seen by jack as a hardware device, hence why `system:capture_1` port is the only system port available.

When jack is configured with the correct hardware device, you should have _at the very least_ `system:capture_1`,  `system:capture_2` (inputs, mics etc) and `system:playback_1`, `system:playback_2` (your speakers or headphone outputs).

Check your jack configuration and make sure it's using the correct sound hardware device. If you're using `qjackctl` to manage `jackd` configuration, you should be able to set the device in that program.

Any other audio you hear is coming from pulseaudio system which is the default, and its a different sound system for linux than jack audio.

Get jack working correctly with your soundcard/audio interface first. Once you can see `system:playback` ports, those are what jamulus program will connect to so you can hear jamulus audio.

**Comment by rdtennent** [comment]
I'm using the built-in sound.  The USB microphone is correctly recognized by qjackctl and Jamulus.  It could be that this  arrangement isn't workable for Jamulus. 

This system was being used because my rpi system stopped working and I haven't received replacements yet.

**Comment by rdica** [comment]
Your mention of "The USB microphone is correctly recognized by qjackctl and Jamulus" tells me that jack is not configured to use your laptop soundcard due to the lack of `system:playback` ports. This indicates jack is using your mic as the default soundcard. Your mic is input only. It provides no playback ports. This is not the device jack needs to use.
Test this by starting without your mic plugged in, take note of what system ports you see in qjackctl or `jack_lsp system`. If you see a pair or more of `system:capture` and `system:playback` ports, then jack is using your laptop soundcard, and you'll be able to connect jamulus outputs to the playback ports and hear others play.

In qjackctl Settings, Advanced, you can set individual Output and Input devices. Try setting Output device to your laptop soundcard. Input is likely already using your mic device. If this works you should then see `system:playback` ports. These are what you'll connect jamulus outputs to.

**Reply by rdica** [reply]
A quick google search of "usb mic jackd" seems to indicate you may have to set up pulseaudio-to-jack bridge and pipe the mic input from pulse out to jamulus in. This also requires that jack is configured to use your laptop soundcard device, not the microphone.

**Comment by rdtennent** [comment]
Yes, without the USB mic plugged in, jack_dsp system is showing capture AND
playback ports.  But I'm going to close this thread.  Thanks for your help
but I only wanted to use this laptop temporarily (until I can restore the
rpi system that I used for years).  It seems too complicated to use this
laptop and a USB mic for Jamulus.
